### DISCUSSION TITLE
Three color problem

### DISCUSSION THREAD

**Discussion by marcelpi** [pmndrs/drei#518] [Q&A] [answered]
Hi there,

I've just noticed that there is some sort of inconsistency on how Three handles colors.

Please check this example:
https://codesandbox.io/s/three-color-problem-3c3b6?file=/src/App.js

Both spheres have the same color assigned: rgb(15,15,255)
One is given a string with the color, the other a Three JS Color object.

Yet somehow the hues rendered in the end are noticeably different.
I can say for sure that the purple hue is wrong(the one using the new THREE.Color() object).

Check the attached picture.
![color](https://user-images.githubusercontent.com/7136479/129459825-b9c4f527-67a6-47c1-b3a8-36b2faaddf93.png)

If this a problem with Three JS handling color or am I missing something?

Any clarification would be greatly appreciated.

**Comment by drcmda** [comment, marked as answer]
three handles colors incorrectly, see https://www.donmccurdy.com/2020/06/17/color-management-in-threejs

r3f uses srgb colorspace and correct gamma automatically.

```jsx
<Canvas linear>
```

if you want back to three default but this is something you should never do, unless you dig the cheap plastic cgi look

**Reply by marcelpi** [reply]
Thanks for the clarification.

So Three does not do a gamma correction when using THREE.Color() and using <Canvas linear> displays all colors without a gamma correction. Did I get this right?

And why is this not handled out of the box by Three JS?
Thanks again.

**Reply by drcmda** [reply]
your discourse thread has more info on it https://discourse.threejs.org/t/three-color-problem/28920/18 it's not 100% clear to my why they don't but it's not as straight forward as it seems.

**Reply by marcelpi** [reply]
Yes, I can see how deep the rabbit hole goes.
Thanks for your input my friend.
